src/makeCalculator.js

```diff
@@ -4,7 +4,51 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  let result = 0;
+
+  function operate(callback, number) {
+    result = callback(result, number);
+
+    return this;
+  }
+
+  function reset() {
+    result = 0;
+
+    return this;
+  }
+
+  function add(a, b) {
+    return a + b;
+  }
+
+  function subtract(a, b) {
+    return a - b;
+  }
+
+  function multiply(a, b) {
+    return a * b;
+  }
+
+  function divide(a, b) {
+    if (b !== 0) {
+      return a / b;
+    } else {
+      throw new Error('Division by zero');
+    }
+  }
+
+  return {
+    get result() {
+      return result;
```

Comment:
The `result` property is being exposed as a getter, which is good practice for encapsulation. However, ensure that there's no way to directly modify the `result` from outside the object, as this would break encapsulation.